### PR TITLE
Adds flip/ for Flax improvement processes.

### PR DESCRIPTION
**Pull request by andsteing**
Adds the directory `flip/` that stores "Flax improvement processes".

Note that the `Type: FLIP` issue is currently called described as `Proposals for breaking or major API changes` - this will be changed to `Flax improvement process` later on to make the use of the word consistent.

### CODE REVIEW

File: flip/README.md
Comment (avital) [inline flip/README.md:33]: Oh? Is this a cool markdown feature I didn't know about -- is the [Flip label] above linked to this?

File: flip/README.md
Comment (avital) [inline flip/README.md:1]:
```suggestion
# FLIP: Flax Improvement Process
```

**Review by avital — COMMENTED** [review]
Thanks for doing this, this is great!

One question about the location. We've been moving a lot of markdown files into docs/ -- should this live there?

(I suppose these really aren't user-facing and should not appear on the navigation bar on flax.readthedocs.org, so maybe markdown files in the main repo are indeed the right place)

File: flip/README.md
Comment (andsteing) [reply flip/README.md:33]: exactly.

**Comment by andsteing** [comment]
> Thanks for doing this, this is great!
> 
> One question about the location. We've been moving a lot of markdown files into docs/ -- should this live there?
> 
> (I suppose these really aren't user-facing and should not appear on the navigation bar on flax.readthedocs.org, so maybe markdown files in the main repo are indeed the right place)

SGTM - moved.

File: docs/flip/README.md
Comment (gnecula) [inline docs/flip/README.md:30]: Some features may have multiple pull requests. Some people may want to include the

Comment (gnecula) [inline docs/flip/README.md:30]: Is it important to have the PR number part of the name? Can it be some other user-chosen name?

Comment (gnecula) [reply docs/flip/README.md:30]: In general, the prescriptions on formatting are pretty precise, and I am wondering which are important and which are not?

Comment (andsteing) [reply docs/flip/README.md:30]: (I used https://github.com/rust-lang/rfcs as a starting point where they also use this scheme)

The `your-flip` was supposed to be the user-chosen name. Clarified.

I thought PR is a useful number, but maybe we would rather want to have shorter IDs only used to differentiate between FLIPs? Updated.

Note that I think it's useful to have a alphabetically increasing list of FLIP filenames. Alternatively we could use the date in YYYYMMDD format - leaving open for further comments.

**Comment by andsteing** [comment]
@avital can you approve once you're happy?
(I'm fine moving it back to the root if you think that's where they should reside...)

**Comment by avital** [comment]
@googlebot I consent.

File: docs/flip/README.md
Comment (avital) [reply docs/flip/README.md]: I like that the name contains the issue number because it makes it easy to find the discussion thread on GitHub later.

**Comment by andsteing** [comment]
@avital @gnecula  PTAL

**Comment by avital** [comment]
Paging this back in real quick -- let's merge it. I was going to ask "why not just use issue templates" but I realize that discussions on those thread get unwieldy quickly. Maybe after merging this it would be a good exercise to look at the issues under the "Improve Linen" and "Improve Flax outside of Linen" milestones and see which of those could be "ported" to this FLIP process.

**Comment by avital** [comment]
Let's run with this as an experiment and see if @jheek likes it. He's my true beta tester of whether processes serve our needs. :)

**Review by avital — APPROVED** [review]

**Pull request merged**
